llvm/include/llvm/CodeGen/TargetInstrInfo.h

```diff
@@ -993,10 +993,16 @@ class TargetInstrInfo : public MCInstrInfo {
   /// The source and destination registers may overlap, which may require a
   /// careful implementation when multiple copy instructions are required for
   /// large registers. See for example the ARM target.
+  ///
+  /// If RenamableDest is true, the copy instruction's destination operand is
+  /// marked renamable.
+  /// If RenamableSrc is true, the copy instruction's source operand is
+  /// marked renamable.
   virtual void copyPhysReg(MachineBasicBlock &MBB,
                            MachineBasicBlock::iterator MI, const DebugLoc &DL,
-                           MCRegister DestReg, MCRegister SrcReg,
-                           bool KillSrc) const {
+                           MCRegister DestReg, MCRegister SrcReg, bool KillSrc,
+                           bool RenamableDest = false,
+                           bool RenamableSrc = false) const {
```

Comment:
Should we just start passing through a SrcFlags/DestFlags instead of unpacking them into kill/renamable?

Reply:
Or pass in actual `MachineOperand`s for src and dest? Or an actual `COPY` instruction?

Reply:
I think passing in MachineOperands would be wrong, you should be able to use this without an existing COPY instruction

Reply:
> I think passing in MachineOperands would be wrong, you should be able to use this without an existing COPY instruction

Agree!

Reply:
> you should be able to use this without an existing COPY instruction

In that case you can create standalone MachineOperands to pass the information into copyPhysReg. I'm just suggesting this as a bit of a hack, because MachineOperand currently does not store its flags in such a way that they can easily be copied in/out of operands and passed around in APIs.

Reply:
Having to construct temporary operands would be worse than just having a bool flag for every bit